pkg/pillar/types/attesttypes.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+//AttestNonce carries nonce published by requester
+type AttestNonce struct {
+	nonce     []byte
+	requester string
+}
+
+//Key returns nonce content, which is the key as well
+func (nonce AttestNonce) Key() []byte {
+	return nonce.nonce
+}
+
+//SigAlg denotes the Signature algorithm in use e.g. ECDSA, RSASSA
+type SigAlg uint8
+
+//CertType carries the certificate use case e.g. ek, ecdh_exchange etc
+type CertType uint8
+
+//Various certificate types published by tpmmgr
+const (
+	EcdsaSha256 SigAlg = iota + 0 // Default
+	RsaRsassa256
+)
+
+//Various types defined under CertType
+const (
+	CertTypeNone CertType = iota + 0 //Default
+	CertTypeOnboarding
+	CertTypeRestrictSigning
+	CertTypeEk
+	CertTypeEcdhXchange
+)
+
+//AttestQuote contains attestation quote
+type AttestQuote struct {
+	nonce     []byte //Nonce provided by the requester
+	sigType   SigAlg //The signature algorithm used
+	signature []byte //ASN1 encoded signature
+	quote     []byte //the quote structure
+}
+
+//Key uniquely identifies an AttestQuote object
+func (quote AttestQuote) Key() []byte {
+	return quote.nonce
+}
+
+//AttestCert contains attest signing certificate published by tpmmgr
+type AttestCert struct {
+	certType CertType
+	cert     []byte //PEM encoded
+}
+
+//Key uniquely identifies an AttestCert object
```

Comment:
Ditto; needs to be usable as a string by the caller.

Reply:
Will you ever have multiple certificates of the same type that you need to tell the controller about?

Reply:
I have introduced UUID as the key for Cert objects.  During rotation, we can have multiple (2 to be exact) certificates of the same type. But UUID will be different. UUID is the key of the object.

Reply:
OK. Is the controller sending the certID in the API?

Reply:
No, this is for internal communication between tpmmgr and zedagent. When zedagent publishes quote as received from tpmmgr, it will either include certHash or certID to identify the certificate to be used for verifying quote signature in the Controller.

Reply:
So will tpmmgr generate a certID?

Reply:
yes, it will.